unstructured/documents/elements.py

```diff
@@ -651,19 +675,35 @@ def to_dict(cls):
 
 
 class Element(abc.ABC):
-    """An element is a section of a page in the document."""
+    """An element is a section of a page in the document.
+
+    There are a few design principles that are followed when creating an element:
+    1. It will always have an ID, which by default is a random UUID.
+    2. Asking for an ID should always return a string, it can never be None.
+    3. When deterministic behavior is needed, the ID can be converted.
+        to a hash based on its text `element.id_to_hash()`
+    4. Even if text attribute is not defined in a subclass, it will default to a blank string.
+    5. Assigning a string ID manually is possible, but is meant to be used
+        only for deserialization purposes.
+    """
 
     text: str
 
     def __init__(
         self,
-        element_id: str | uuid.UUID | NoID | UUID = NoID(),
+        element_id: str | NoID = NoID(),
```

Comment:
If the purpose of the `NoID` class is to be a placeholder indicating we want the `Element` to self-assign an id, do we need the class at all? Could we just use `None` instead (i.e. `element_id` could have type `Optional[str]`)? Or do we need `NoID` for backward compatibility with code out in the wild (in which case maybe we could deprecate)?

Reply:
I think that's an excellent question @qued. I'm inclined to agree that we should get rid of `NoID` and fix up whatever internal calls we have in a partitioner that might use it.

It doesn't add any new information, at least not anymore. I think at one time you could "pre-select" the "`id_type`" by specifying either `UUID()` (not the stdlib `uuid.UUID`) or `NoID()`, but that's no longer the case and I think this should go away now and just use `None` as the default.